### PR TITLE
Hotfix: Merge hotfixes from 4.2.1.

### PR DESCRIPTION
**Pull request by guddo**

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3b6d2c743c330ecbf4d7a5529c6d2117e304460a
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: be1844c1ea7fd4a9cffffcf8a3a64aba2ddf6d64
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 06b21d02cbca48a83297fa695f917b37da587313
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 72a607fdbb8a0e1c9a0cde82adfad0b135e8a42d
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 20e6c430d6dd66316298a2ac0acc86ba52ae16d8
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3b6d2c743c330ecbf4d7a5529c6d2117e304460a
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: be1844c1ea7fd4a9cffffcf8a3a64aba2ddf6d64
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 06b21d02cbca48a83297fa695f917b37da587313
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 72a607fdbb8a0e1c9a0cde82adfad0b135e8a42d
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 6dda0814b54ce61f6dcb8a8a3af0a726d5e72075
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5a15aae43952eab395bcf58e34c5fb016da980e1
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 120aafddf99a54ec055ebf8d3f613907406970cb
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 91ee3fde24823c3ae0df238be25bd69771e998c4
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 8d32c754d7d56373fd210d87fde218da6489680d
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5bd500e06236601e81d92212de4f66c06f335a5c
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 69b1154192c424773935e5b7473c282f0eb2bcc4
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 13390465c84b425c3607265581a05066130b26b1
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

File: modules/ding_loan/plugins/content_types/loans.inc [hunk 1]
```diff
@@ -51,6 +51,14 @@ function ding_loan_loans_content_type_render($subtype, $conf, $panel_args, $cont
 
       $ids[] = $item->ding_entity_id;
     }
+    ding_entity_load_multiple($ids);
```
Comment (guddo) [inline modules/ding_loan/plugins/content_types/loans.inc]: These changes I am uncertain about.

File: modules/ding_loan/plugins/content_types/loans.inc [hunk 2]
```diff
@@ -168,10 +176,13 @@ function ding_loan_loans_form($form, &$form_state, $account, $loans = array()) {
       // about the associated material.
       /** @var \TingEntity $entity */
       $entity = $loan->entity;
+
       if (!is_object($entity) || NULL === $entity->getTingObject()) {
         // Patron have this item in loans list, but the search provider can't
-        // find this record for current agencyId, so we load it separately.
-        $ting_object = ting_search_object_load($loan->ding_entity_id);
+        // find this record for current agencyId, so ask ding_provider for help:
```
Comment (guddo) [inline modules/ding_loan/plugins/content_types/loans.inc]: These changes I am uncertain about.

File: modules/ding_provider/ding_provider.module [hunk 3]
```diff
@@ -592,15 +592,8 @@ function ding_provider_function($type, $hook, $quiet = FALSE) {
  *   Ting entity object based on the entity ID given.
  */
 function ding_provider_get_pseudo_entity($ding_entity_id) {
-  $local_id = drupal_substr($ding_entity_id, strpos($ding_entity_id, ':') + 1);
```
Comment (guddo) [inline modules/ding_provider/ding_provider.module]: These changes I am uncertain about.

File: modules/ding_reservation/ding_reservation.module [hunk 4]
```diff
@@ -516,9 +516,12 @@ function ding_reservation_reservations_form($form, &$form_state, $reservations,
     /** @var \TingEntity $entity */
     $entity = $reservation->entity;
     if (!is_object($entity) || NULL === $entity->getTingObject()) {
-      $entity = ding_provider_get_pseudo_entity($reservation->ding_entity_id);
-      // Don't link the title as the data well do not known this title. It may
-      // be a inter library loan.
+      // If the provider wasn't able to fetch an object, try again without local
```
Comment (guddo) [inline modules/ding_reservation/ding_reservation.module]: These changes I am uncertain about.

File: modules/ding_loan/plugins/content_types/loans.inc (hunk 1 above)
Comment (guddo) [reply modules/ding_loan/plugins/content_types/loans.inc]: @AnatolyMuntean Can you verify wether these changes are valid or not?

File: modules/ding_loan/plugins/content_types/loans.inc (hunk 2 above)
Comment (guddo) [reply modules/ding_loan/plugins/content_types/loans.inc]: Is this needed after code was changes from $entity to $ting_object?

Comment (guddo) [reply modules/ding_loan/plugins/content_types/loans.inc]: - Original change: https://github.com/easySuite/ding2/pull/423/files#diff-6023214c29e0026da2b604ef2d5703aeL183
- Changes that was done by others: https://github.com/ding2/ding2/commit/0448d72e686685c780de74f9957149a3cd273898
- Compare our 4.2.1 with 4.4.2: https://github.com/easySuite/ding2/compare/hotfix-4.2.1...development-4.4.x#diff-6023214c29e0026da2b604ef2d5703aeL170

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 13390465c84b425c3607265581a05066130b26b1
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 4d9f0d99296bea77f5e7638a6fc5356f70e064c5
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 13390465c84b425c3607265581a05066130b26b1
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 4d9f0d99296bea77f5e7638a6fc5356f70e064c5
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

File: modules/ding_loan/plugins/content_types/loans.inc [hunk 5]
```diff
@@ -181,32 +192,25 @@ function ding_loan_loans_form($form, &$form_state, $account, $loans = array()) {
         }
       }
 
-      if (is_object($entity) && NULL !== $entity->getTingObject()) {
```
Comment (guddo) [inline modules/ding_loan/plugins/content_types/loans.inc]: - Original change: https://github.com/easySuite/ding2/pull/423/commits/3c5967e328b77dc03a79e13527288c1fd5b438bf#diff-6023214c29e0026da2b604ef2d5703aeL192

**Comment by guddo** [comment]
Also there is conversation about this in another PR: https://github.com/easySuite/ding2/pull/464/files

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 0e0ed7aac2be60bef10fc3554165190512c56036
  * Commits must be in the following format: [issue ID] - [short summary of change]

 * Commit: 2171551b2adbd75adec3e16eb3d375335476bfe8
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 1dc6df3aa457b727a5ca01035621e4383bec7f72
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 1dc6df3aa457b727a5ca01035621e4383bec7f72
  * Commits must be in the following format: [issue ID] - [short summary of change]

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Pull request merged**